UserTools/nhits/nhits.cpp

```diff
@@ -119,38 +138,75 @@ void NHits::AlgNDigits(const SubSample * sample)
   int first_digit_in_window = 0;
   for(;current_digit < ndigits; ++current_digit) {
     // Update first digit in trigger window
-    while(sample->m_time[first_digit_in_window] < sample->m_time[current_digit] - m_trigger_search_window){
-      ++first_digit_in_window;
+
+    if( !m_degrade_data ){
+      TimeDelta::short_time_t digit_time = sample->m_time.at(current_digit);
+      while(TimeDelta(sample->m_time[first_digit_in_window]) < TimeDelta(digit_time) - m_trigger_search_window){
+	++first_digit_in_window;
+      }
+    }else{
+      //F. Nova degrade info from float to int to match GPU run
+      int digit_time = (int)sample->m_time.at(current_digit);
+      while(TimeDelta(sample->m_time[first_digit_in_window]) < TimeDelta(digit_time) - m_trigger_search_window){
```

Comment:
You still only degrade the trigger time, but not the times of the other digits.

```suggestion
      while(TimeDelta((int) sample->m_time[first_digit_in_window]) < TimeDelta(digit_time) - m_trigger_search_window){
```

To make this the same as the GPU version, you might also have to change the `<` to `<=`, depending on how exactly that works.

Reply:
done